python-lib/osisoft_client.py

```diff
@@ -896,6 +906,103 @@ def close(self):
     # }
 
 
+class OSIsoftBatchWriter(object):
+    # Each row of data added (write_row) first goes in a streak buffer
+    # streak buffer is meant to be used to push a flow of values to one AF path / webid
+    # If the new row concerns another webid, the current streak buffer is flushed into the request buffer
+    # When request buffer is full, it is flushed into the batch endpoint
+
+    # Each write_row call is adds an entry to the responses list
+    # The pointer into that list is past as a _dku_counter parameter, extracted from the requests just before sending the batch
+    # Uppon receiving the batch response, each individual reponse is reordered
+    #   and the status code / error messages are stored into the responses list at the right pointer
+
+    # Possible improvement: flush the responses list (and result writing in the recipe.py) at each _flush_requests to keep memory from going up.
+
+    # My Regards to the unsung Hero who had to review this code
+
+    def __init__(self, client, max_streak_buffer_size=500, max_requests_buffer_size=500):
+        logger.info("Initializing OSIsoftBatchWriter, msbs={}, mrbs={}".format(max_streak_buffer_size, max_requests_buffer_size))
+        self.client = client
+        self.streak_buffer = []     # list of points from a single webid -> can be all sent in one request
+        self.requests_buffer = []   # list of independant requests -> sent one batch at a time
+        self.responses = []         # building a list of reponses in same order as write_row calls
+        self.current_webid = None
+        self.max_buffer_size = max_streak_buffer_size
+        self.max_requests_buffer_size = max_requests_buffer_size
+        self.current_streak = 0
+        self.row_number = 0
+
+    def write_row(self, webid, timestamp, value):
+        response = None
+        if not validate_timestamp(timestamp):
+            error_message = "Timestamp '{}' has an invalid format".format(
+                    timestamp
+            )
+            logger.error(error_message)
+            # No valid timestamp so we skip this row
+            return {
+                "Error": error_message
+            }
+        # mark in self.responses that status of this write depends on result of streak X
+        if self.current_webid is None:
+            logger.info("webid now is {}".format(webid))
```

Comment:
format a lil bit 
also, do we want to show every webid? 
```suggestion
            logger.info("webid : {}".format(webid))
```

Reply:
👍 done in [a40801f](https://github.com/dataiku/dss-plugin-osisoft/pull/73/commits/a40801ffaaa98f69f01ec538e3a90a248122715c)